### PR TITLE
ROC-7708: Replace exchange token endpoint with oidc endpoint

### PR DESCRIPTION
**Pull request by sabahirfan**
### JIRA link (if applicable) ###

[ROC-7708](https://tools.hmcts.net/jira/browse/ROC-7708)

### Change description ###
**This is merged into PR https://github.com/hmcts/cmc-claim-store/pull/1598** No need to merge this one to master.

Remove calls to /oauth2/* endpoints and use new openid endpoints
/oauth2/token is used to `exchangeCode` remove all the calls to this endpoint. Use /o/token endpoint  instead.

**Does this PR introduce a breaking change?** (check one with "x")

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/UserServiceTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=5208907&bid=17082277)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/UserServiceTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=5208907&bid=17082277)

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/UserServiceTest.java [hunk 1]
```diff
@@ -71,15 +83,58 @@ public void findsUserInfoForAuthToken() {
     @Test
     @Ignore("Enable this when using /o/userinfo")
```
Comment (andrewwa-kainos) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/services/UserServiceTest.java:84]: @sabahirfan  Can this ignore be removed now?

**Comment by sunilkumar-cts** [comment]
Will be reopened

**Pull request closed without merging**